### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by samdenty**
This PR makes the documentation flow more consistent for newcomers, and

- Renames `GqlessConfig` to `GQlessConfig` (so it's consistent with the new logo)
- Renames `gqlessError` to `GQlessError`
- Removes the `endpoint` option from the configuration, and instead always defaults to introspection one
  - It's confusing why theres two of them, and the user can change it later by modifying the file anyway

### CODE REVIEW

**Comment by PabloSzx** [comment]
I don't agree with GPL v3 license, I don't want to restrict gqless usage to open-source

**Comment by samdenty** [comment]
Doesn't prevent use of GQless in non-opensource, just that changes to the core should be open-sourced (the original license of the project)

File: docs/core/basic-usage.mdx [hunk 1]
```diff
@@ -84,36 +108,3 @@ async function Example() {
...
-    };
-  }
-  return null;
-});
-
-// Later...
-
-// It will refetch 'id' & 'name' of 'query.user'
-const refetchedUser = await refetch(query.user);
-```
```
Comment (PabloSzx) [inline docs/core/basic-usage.mdx:119]: why?

Comment (samdenty) [reply docs/core/basic-usage.mdx:119]: pr is still a draft! 😃

Comment (samdenty) [reply docs/core/basic-usage.mdx:119]: but I moved that bit further up ^

File: packages/gqless/src/Error/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import type { GraphQLError } from 'graphql';
 
-export class gqlessError extends Error {
+export class GQlessError extends Error {
```
Comment (PabloSzx) [inline packages/gqless/src/Error/index.ts:3]: The convention I was indirectly using was that Types start with uppercase, and values start with lower-case, and also changing this class is a breaking change, but if you really want to change it,  it should be also provided the previous class with the `@deprecated` comment that points to the same class

File: packages/cli/src/config.ts [hunk 3]
```diff
@@ -24,21 +24,22 @@ function isStringRecord(v: unknown): v is Record<string, string> {
   );
 }
 
-export const defaultConfig = {
-  endpoint: '/api/graphql',
-  enumsAsStrings: false,
+export const DUMMY_ENDPOINT = 'SPECIFY_ENDPOINT_OR_SCHEMA_FILE_PATH_HERE';
+
+export const defaultConfig: Required<GQlessConfig> = {
```
Comment (PabloSzx) [inline packages/cli/src/config.ts:29]: this default config type (Required\<GQlessConfig\>) doesn't take in account the deep fields in the introspection config

File: packages/cli/src/config.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import { resolve } from 'path';
 import type { GenerateOptions } from './generate';
 import type { IntrospectionOptions } from './introspection';
 
-export type GqlessConfig = GenerateOptions & {
+export type GQlessConfig = Omit<GenerateOptions, 'endpoint'> & {
```
Comment (PabloSzx) [inline packages/cli/src/config.ts:6]: This type name change is a breaking change and the previous version should be also provided with the `@deprecated` comment

File: packages/cli/src/generate.ts [hunk 5]
```diff
@@ -28,23 +28,31 @@ import { codegen } from '@graphql-codegen/core';
...
+   *
+   * @example
+   * ```json
+   * scalarTypes: {
+   *   DateTime: "string",
+   * }
+   * ```
    */
-  preImport?: string;
+  scalarTypes?: Record<string, string>;
```
Comment (PabloSzx) [inline packages/cli/src/generate.ts:51]: This type name change is a breaking change and the previous version should be also provided with the `@deprecated` comment

File: packages/cli/src/generate.ts [hunk 6]
```diff
@@ -138,7 +150,7 @@ export async function generate(
           onlyOperationTypes: true,
           declarationKind: 'interface',
           addUnderscoreToArgsType: true,
-          scalars,
+          scalars: scalarTypes,
```
Comment (PabloSzx) [inline packages/cli/src/generate.ts:153]: if you want to change this name, why this alias?

File: packages/cli/src/config.ts [hunk 7]
```diff
@@ -217,15 +217,20 @@ export const gqlessConfigPromise: Promise<{
                 const { format } = (await import('./prettier')).formatPrettier({
                   parser: 'typescript',
                 });
+
+                const config: GQlessConfig = { ...defaultConfig };
+                delete config.preImport;
+                delete config.enumsAsStrings;
```
Comment (PabloSzx) [inline packages/cli/src/config.ts:223]: can you share why are you making this change?

File: packages/gqless/src/Client/client.ts [hunk 8]
```diff
@@ -143,14 +143,14 @@ export interface Mutate<
...
-        error: gqlessError,
+        error: GQlessError,
         helpers: MutateHelpers<GeneratedSchema>
       ) => void;
     }
   ): Promise<T>;
 }
 
-export interface GqlessClient<
+export interface GQlessClient<
```
Comment (PabloSzx) [inline packages/gqless/src/Client/client.ts:153]: This type name change is a breaking change and the previous version should be also provided with the `@deprecated` comment

File: internal/website/sidebars/react.js [hunk 9]
```diff
@@ -1,45 +1 @@
...
-      'react/api/interfaces/userefetch',
-      'react/api/interfaces/userefetchoptions',
-      'react/api/interfaces/userefetchstate',
-      'react/api/interfaces/usesubscription',
-      'react/api/interfaces/usetransactionquery',
-      'react/api/interfaces/usetransactionquerystate',
-    ],
-  },
-];
+module.exports = [];
```
Comment (PabloSzx) [inline internal/website/sidebars/react.js:1]: why are you removing the React API Reference

File: internal/website/src/pages/examples/index.tsx [hunk 10]
```diff
@@ -45,11 +45,7 @@ export default function Page() {
           usages of <b>gqless</b>.
         </Text>
         <Text>
-          And we encourage you to{' '}
-          <Link to="https://discord.gg/FjwyGQKYER">
-            Join our Discord server
-          </Link>{' '}
-          or open a new{' '}
```
Comment (PabloSzx) [inline internal/website/src/pages/examples/index.tsx:52]: Why are you removing Discord?

File: internal/website/sidebars.js [hunk 11]
```diff
@@ -1,47 +1,73 @@
...
-  },
+  docs: [
+    {
+      type: 'category',
+      label: 'Introduction',
+      collapsed: false,
+      items: [
+        'intro',
+        'intro/how-it-works',
+        // 'introduction/features',
```
Comment (PabloSzx) [inline internal/website/sidebars.js:10]: why are you removing the features page from the index but keeping the doc itself?

**Comment by PabloSzx** [comment]
Why there is the ci/codesandbox blocking and saying it failed? codesandbox is not compatible with pnpm / monorepo

**Review by PabloSzx — CHANGES_REQUESTED** [review]

File: packages/gqless/src/Error/index.ts (hunk 2 above)
Comment (samdenty) [reply packages/gqless/src/Error/index.ts:3]: We have no users yet, so this shouldn't be an issue.

File: internal/website/sidebars.js (hunk 11 above)
Comment (samdenty) [reply internal/website/sidebars.js:10]: going to update homepage later

File: README.md [hunk 12]
```diff
@@ -9,9 +9,6 @@
   <a href="https://gqless.com">
     <img alt="Documentation" src="https://img.shields.io/badge/documentation-documentation?color=C00B84">
   </a>
-  <a href="https://discord.gg/9DNNPedhbx">
-    <img alt="Discord" src="https://img.shields.io/discord/793628688439836694?color=7289d9&label=discord">
-  </a>
```
[the remark this answers is not in the document]
Comment (samdenty) [reply README.md:14]: switching to discussions

File: packages/cli/src/config.ts (hunk 7 above)
Comment (samdenty) [reply packages/cli/src/config.ts:223]: default config doesn't need these, advanced usage only

**Pull request merged**